### ISSUE TITLE
Duplications don't save with ExpAdv2 Gate

### ISSUE THREAD

**Issue by Garrus007** [Rusketh/ExpAdv2#296]
If you make an object with ExpAdv2 Gate, you won't able to save It's dublication. There is nothing happens when you click "Save dupe" button in "Dupe" tab.
1. Make any object with ExpAdv2 Gate
2. Take Duplication tool and left click to the object. Right click somewhere will check, that the object duplicated correctly
3. Open menu (press Q) and go to the "Dupes" tab
4. Key "Save Dupe" is active. Press it. Menu will close.
5. Go to the dupes tab again. In "My Dupes" section no new dupe. Button "Save Dupe" isn't active. It will be active again if you left click you object with duplicator again.

**Comment by VINTproYKT** [comment]
> 2. Take Duplication tool and left click to the object. Right click somewhere will check, that the object duplicated correctly

Are you sure you doing it right way? First right click on contraption, then left click somewhere to check.

Hm, same issue was with my contraption, I don't ever remember what was it, but after pressing "Save Dupe" nothing saved.

**Comment by bigdogmat** [comment]
I know the default gmod dupe has a lot of issues, so it could be it's disabled. Though it should just not dupe the chip and save the rest.

**Comment by Garrus007** [comment]
Yes, I check it, I do it right. When I save a dupe **without** any WireMod parts or with **Expression 2 gate** It is OK. So, I think the problem, that it can't save chip.

**Comment by VINTproYKT** [comment]
Yeah, standard duplicator looks dumb for some developers that stucked with this. But in case with Expression Advanced 2 that was made to be independent-some, it's needed to integrate gate with standard duplicator.

**Comment by bigdogmat** [comment]
It's not that it looked dumb, it's that it crashed a shit ton and did weird shit.

**Comment by Garrus007** [comment]
I've test it with Advanced Duplicator. It doesn't work. Dupe the object with Advanced duplicator, go to menu, click "save", type dupe name  - nothing happens. When saving object without ExpAdv2 gate  - all works, when you click "save" you can see shortly loading bar "saving" and dupe avaliable in dupe list.

**Comment by VINTproYKT** [comment]
I made research on it *in singleplayer*.

1. In case with EA2 gates I don't get any error and can paste with both duplicators (native GMod duplicator, WireTeam's [Advanced Duplicator from Steam Workshop](http://steamcommunity.com/sharedfiles/filedetails/?id=163806212)).
2. Same with EA2 screens.
3. In case with EA2 bots it's bit different: I need to catch duplication exactly **before** first bot spawned will be initialized, **or** it will cause console `[ERROR]` while trying to paste (depends on coroutine used in `behavior`).

**Comment by Rusketh** [comment]
The advanced duplicator should be working unless wire has made any changes to it recently. As for the stock duplicator tool, this will not be supported ever. This is due to the fact that Garry's duplicator does not support the functionality needed to make expadv2 compatible.

Can somebody please double check to see if this issue is with the advanced duplicator?

**Comment by Garrus007** [comment]
I did the tests. It works in single player mode and and in multiplayer mode too (**wtf?!**).
But if I save a dupe in singleplyer and try to load it in multiplayer It'll nothing paste (I do the click - hear sound, but nothing happens). If I save dupe in multiplayer and try to paste it - code is broken, only 1 first line

**Comment by VINTproYKT** [comment]
We need new duplicator :smile:
